### PR TITLE
refactor/ipgeo_from_backend

### PR DESCRIPTION
**Pull request by JarbasAl**
use https://github.com/OpenVoiceOS/ovos-backend-client/pull/27

### CODE REVIEW

File: ovos_phal_plugin_ipgeo/__init__.py [hunk 1]
```diff
@@ -34,25 +36,11 @@ def on_reset(self, message=None):
...
-        return {"city": city_data,
-                "coordinate": coordinate_data,
-                "timezone": timezone_data}
-
-
+        try:
+            # configured backend may throw some errors if its down
+            api = GeolocationApi()
+            return api.get_ip_geolocation(ip)
+        except:
```
Comment (NeonDaniel) [inline ovos_phal_plugin_ipgeo/__init__.py:43]: This should probably log the raised exception for debugging backend problems

**Review by NeonDaniel — CHANGES_REQUESTED** [review]

File: ovos_phal_plugin_ipgeo/__init__.py [hunk 2]
```diff
@@ -57,25 +59,12 @@ def on_reset(self, message=None):
...
-        return {"city": city_data,
-                "coordinate": coordinate_data,
-                "timezone": timezone_data}
-
-
+        try:
+            # configured backend may throw some errors if its down
+            api = GeolocationApi()
+            return api.get_ip_geolocation(ip)
+        except Exception as e:
```
Comment (mikejgray) [inline ovos_phal_plugin_ipgeo/__init__.py:66]: For future dev efforts, we should log the exception, so we can be more aware of what issues may come up and handle them properly

File: ovos_phal_plugin_ipgeo/__init__.py [hunk 3]
```diff
@@ -57,25 +59,12 @@ def on_reset(self, message=None):
...
+            LOG.exception("Backend Geolocation API error!")
```
Comment (mikejgray) [inline ovos_phal_plugin_ipgeo/__init__.py:67]: ...unless that's what `LOG.exception()` does, grabbing that exception :)

File: ovos_phal_plugin_ipgeo/__init__.py [hunk 4]
```diff
@@ -57,25 +59,12 @@ def on_reset(self, message=None):
...
+            # force offline backend api (direct call)
+            api = GeolocationApi(backend_type=BackendType.OFFLINE)
+            return api.get_ip_geolocation(ip)
```
Comment (mikejgray) [inline ovos_phal_plugin_ipgeo/__init__.py:70]: https://github.com/OpenVoiceOS/ovos-backend-client/blob/bad1b634340ac379e0aa8451f8588306058eec93/ovos_backend_client/backends/offline.py#L294

Wrong method name

**Review by mikejgray — CHANGES_REQUESTED** [review]

File: ovos_phal_plugin_ipgeo/__init__.py (hunk 3 above)
Comment (JarbasAl) [reply ovos_phal_plugin_ipgeo/__init__.py:67]: yup, LOG.exception includes the stack trace

File: ovos_phal_plugin_ipgeo/__init__.py (hunk 4 above)
Comment (JarbasAl) [reply ovos_phal_plugin_ipgeo/__init__.py:70]: correct api is https://github.com/OpenVoiceOS/ovos-backend-client/blob/bad1b634340ac379e0aa8451f8588306058eec93/ovos_backend_client/api.py#L358

**Review by mikejgray — APPROVED** [review]

**Pull request merged**